components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -16,27 +16,46 @@ import mozilla.components.lib.state.Action
  * [Action] implementation related to [BrowserState].
  */
 sealed class BrowserAction : Action
+
 /**
  * [BrowserAction] implementations related to updating the list of [TabSessionState] inside [BrowserState].
  */
 sealed class TabListAction : BrowserAction() {
     /**
      * Adds a new [TabSessionState] to the list.
+     *
+     * @property tab the [TabSessionState] to add
+     * @property select whether or not to the tab should be selected.
+     * @property parent the parent [TabSessionState] of the provided tab. The
+     * tab will be added next to its parent, if provided.
      */
-    data class AddTabAction(val tab: TabSessionState, val select: Boolean = false) : TabListAction()
+    data class AddTabAction(
+        val tab: TabSessionState,
+        val select: Boolean = false,
+        val parent: TabSessionState? = null
+    ) : TabListAction()
 
     /**
      * Marks the [TabSessionState] with the given [tabId] as selected tab.
+     *
+     * @property tabId the ID of the tab to select.
      */
     data class SelectTabAction(val tabId: String) : TabListAction()
 
     /**
      * Removes the [TabSessionState] with the given [tabId] from the list of sessions.
+     *
+     * @property tabId the ID of the tab to remove.
+     * @property selectParentIfExists whether or not a parent tab should be
+     * selected if one exists.
      */
-    data class RemoveTabAction(val tabId: String) : TabListAction()
+    data class RemoveTabAction(val tabId: String, val selectParentIfExists: Boolean = false) : TabListAction()
```

Comment:
I wonder why we decided to use `false` as default in `SessionManager` (maybe to not change the default behavior?). It seems like `true` is the default you'd expect in a browser.

Reply:
Yeah, makes sense. I think we wanted to preserve the default behaviour at the time but we don't need to do this now. Updated.